### PR TITLE
Move protocol implementation into new package

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

### CODE REVIEW

**Comment by testforstephen** [comment]
This PR is related to https://github.com/Microsoft/java-debug/issues/50. Move it to a new package, but still keep it in core.jar.

**Comment by akaroml** [comment]
Let's release this in 0.3.0.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    }
+
+    /**
+     * Sets terminateSession flag to true. And the dispatcher loop will be terminated after current dispatching operation finishes.
+     */
+    public void stop() {
+        this.terminateSession = true;
+    }
+
+    public boolean isTerminating() {
```
Comment (yaohaizh) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java]: Quite curious about the different about 
```isTerminating``` and "terminateSession```

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    protected void sendMessage(Messages.ProtocolMessage message) {
+        message.seq = this.sequenceNumber.getAndIncrement();
+
+        String jsonMessage = JsonUtils.toJson(message);
+        byte[] jsonBytes = jsonMessage.getBytes(PROTOCOL_ENCODING);
+
+        String header = String.format("Content-Length: %d%s", jsonBytes.length, TWO_CRLF);
+        byte[] headerBytes = header.getBytes(PROTOCOL_ENCODING);
+
+        byte[] data = new byte[headerBytes.length + jsonBytes.length];
```
Comment (yaohaizh) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java]: Sounds some duplication? Should be ByteBuffer?

Comment (testforstephen) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java]: Nice catch. fixed.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java (hunk 1 above)
Comment (testforstephen) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java]: duplicate. Remove `isTerminating` method.

**Pull request merged**